src/ray/core_worker/core_worker.cc

```diff
@@ -1217,6 +1216,13 @@ void CoreWorker::Exit(
     // drain the object references.
     task_execution_service_.post(
         [this, shutdown]() {
+          RAY_LOG(INFO) << "Wait for currently executing tasks in the underlying thread "
+                           "pools to finish.";
+          // Wait for currently executing tasks in the underlying thread pools to
+          // finish. Note that if tasks have been posted to the thread pools but not
+          // started yet, they will not be executed.
```

Comment:
Please correct my understanding if wrong: we stop accepting new tasks in the call above when we do `task_execution_service_.stop();`, right?

So after that call, no more callbacks will be accepted on the `task_execution_service_`, then after this call to `task_receiver_->Stop()` we know that all ongoing work has finished.